src/main/cpp/apple_fsnotifier.cpp

```diff
@@ -166,6 +189,7 @@ Java_net_rubygrapefruit_platform_internal_jni_OsxFileEventFunctions_stopWatch(JN
     // TODO: consider using FSEventStreamFlushSync to flush all pending events.
     watcherStream = NULL;
 
+    // TODO Shouldn't we stop this first and then release the rest?
```

Comment:
yeah, I would stop all the work first and then free everything after

Reply:
Fixed.